hooks/pre/13-dns_check.rb

```diff
@@ -0,0 +1,21 @@
+# Check to verify forward/reverse dns matches hostname
+
+if app_value(:disable_system_checks)
+  logger.warn 'Skipping system checks.'
+else
+  hostname = `$(hostname -f)`
+  ip = getent hosts `#{hostname} | awk '/STREAM { print $1 }'`
+  fqdn = `getent hosts #{ip} | awk '{ print $2 }'`
+  if hostname == fqdn
```

Comment:
This is more of a general style and maybe my python side shining through. From the [Zen of Python](http://legacy.python.org/dev/peps/pep-0020/): ```Flat is better than nested```. More concrete I think the general flow is easier to understand when it's flat. In the following code I also added another check and rephrased it a bit. This is untested.

```ruby
ip = `getent hosts #{hostname} | awk '/STREAM { print $1 }'`
if ip.empty?
  logger.error "Unable to resolve forward DNS for '#{hostname}'"
  kafo.class.exit(1)
end

`ip a | grep -q #{ip}`
unless $?.success?
  logger.error "Forward DNS points to '#{ip}' which is not configured on this server"
  kafo.class.exit(1)
end

reverse = `getent hosts #{ip} | awk '{ print $2 }'`
unless hostname == reverse
  logger.error "Reverse DNS '#{reverse}' does not match hostname '#{hostname}'"
  kafo.class.exit(1)
end
```